Add git info as PDF revision in footnote

MG-5 opened the pull request:
Mit dem PR  wird beim Builden der PDFs (von Github Action)  die jeweiligen git tags und commit hash in die Fußzeile gesetzt, sodass man nachvollziehen kann, dass stets die aktuelle Version genutzt wird.

So sieht es dann aus:

<img width="1889" height="249" alt="image" src="https://github.com/user-attachments/assets/623c2cd9-78f6-4f5d-bfeb-2109abb87aca" />

Es wurde hier erfolgreich getestet: https://github.com/MG-5/gitinfo-in-footnote

MG-5 committed:
Add git commit hash into footnote at bulding PDFs by Github Action

MG-5 committed:
Add git tag

LeSpocky reviewed (commented):
Grundsätzlich finde ich das eine sehr gute Idee. 🙂 

Vorschlag/Frage zur Ausgabe: Ausgabe wäre mit dem PR Stand jetzt hier direkt auf der Version v5.0.1 folgende: "v5.0.1 (git-c368c34)".  Wäre man zwei Commits weiter wäre die Ausgabe: "v5.0.1 (git-37ffde0)".  Man kann also nicht auf den ersten Blick unterscheiden ob man genau auf dem Tag ist oder einige Commits weiter.

Man könnte auch direkt den Output von `git describe --tags` ohne die Option `--abbrev` nehmen.  Dann wäre die Ausgabe "v5.0.1" wenn man direkt auf dem Tag ist und "v5.0.1-2-g37ffde0" für "zwei Commits weiter".  Das wäre die Variante, die ich persönlich bevorzugen würde, man hätte dann den Git Hash nur dabei, wenn man nicht direkt auf einem Tag ist.

Evtl. ginge auch die Option `--long` wenn man immer den Hash dabei haben will.  Dann wäre es "v5.0.1-0-gc368c34" gegenüber "v5.0.1-2-g37ffde0".

Davon ab ist jede der Varianten ist besser als der aktuelle Stand. 👍🏼

MG-5 commented:
Daran dachte ich auch schon. Dein Lösungsvorschlag ist mir nicht eingefallen, der ist sehr gut :D übernehme ich gleich so

MG-5 committed:
Use another syntax for revision:
Shows v1.0.0 if exactly on tagged commit
Shows v1.0.0-xxxxxx if n commits after tagged commit

LeSpocky commented:
Ich sehe gerade die Änderung an *.github/workflows/ci.yml* brauchen wir in *.github/workflows/release.yml* nochmal. :face_with_peeking_eye: :sweat_smile:

penguineer commented:
> Man könnte auch direkt den Output von `git describe --tags` ohne die Option `--abbrev` nehmen. Dann wäre die Ausgabe "v5.0.1" wenn man direkt auf dem Tag ist und "v5.0.1-2-g37ffde0" für "zwei Commits weiter". Das wäre die Variante, die ich persönlich bevorzugen würde, man hätte dann den Git Hash nur dabei, wenn man nicht direkt auf einem Tag ist.

Dann muss die Checkout-Action so angepasst werden, dass alle Commits ausgecheckt werden, weil sonst der Tag nicht gefunden werden kann.

LeSpocky commented:
> Dann muss die Checkout-Action so angepasst werden, dass alle Commits ausgecheckt werden, weil sonst der Tag nicht gefunden werden kann.

Das passiert  ja explizit über `run: git fetch --prune --unshallow --tags`, aber nur für *ci.yml* und aktuell nicht für *release.yml* … ggf. reicht aber auch folgendes[^1]:

```yaml
- uses: actions/checkout@v5
  with:
    fetch-tags: true
```

[^1]: https://github.com/actions/checkout#usage

MG-5 commented:
Ich werde noch ein PR machen, wo Release zum Bauen dann direkt auf Action von CI zugreift. Dann ist das Bauen von PDFs nicht mehr redundant.

MG-5 commented:
> Ich werde noch ein PR machen, wo Release zum Bauen dann direkt auf Action von CI zugreift. Dann ist das Bauen von PDFs nicht mehr redundant.

Siehe PR #20

MG-5 commented:
> ggf. reicht aber auch folgendes[1](#user-content-fn-1-d0c289f29e4d82e2763f4c5aad51e242):
> 
> ```yaml
> - uses: actions/checkout@v5
>   with:
>     fetch-tags: true
> ```

Ich habe von Bugs bzw error reports gelesen, dass das nicht funktioniert und als Workaround  wurde das vorgeschlagen, was ich bereits habe (`git fetch --prune --unshallow --tags`). Das lasse ich jetzt mal so
